test/functional/Specification/SpecificationLoaderTest.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OnMoon\OpenApiServerBundle\Test\Functional\Specification;
+
+use OnMoon\OpenApiServerBundle\Specification\Definitions\Specification;
+use OnMoon\OpenApiServerBundle\Specification\Definitions\SpecificationConfig;
+use OnMoon\OpenApiServerBundle\Specification\SpecificationLoader;
+use OnMoon\OpenApiServerBundle\Specification\SpecificationParser;
+use OnMoon\OpenApiServerBundle\Types\ScalarTypesResolver;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Contracts\Cache\ItemInterface;
+use Symfony\Contracts\Cache\TagAwareCacheInterface;
+use Throwable;
+
+use function Safe\rename;
+use function Safe\sprintf;
+
+/**
+ * @covers \OnMoon\OpenApiServerBundle\Specification\SpecificationLoader
+ */
+class SpecificationLoaderTest extends TestCase
+{
+    protected SpecificationLoader $specificationLoader;
+
+    public const SPECIFICATION_NAME                    = 'cards';
+    public const SPECIFICATION_LOADER_CACHE_KEY_PREFIX = 'openapi-server-bundle-specification-';
+
+    public function setUp(): void
+    {
+        $this->specificationLoader = new SpecificationLoader(
+            new SpecificationParser(new ScalarTypesResolver()),
+            new FileLocator(),
+            new class () implements TagAwareCacheInterface {
+                /** @var Specification[] $specs  */
+                private array $specs = [];
+
+                /**
+                 * {@inheritDoc}
+                 */
+                public function get(string $key, callable $callback, ?float $beta = null, ?array &$metadata = null)
+                {
+                    Assert::assertSame(SpecificationLoaderTest::SPECIFICATION_LOADER_CACHE_KEY_PREFIX . SpecificationLoaderTest::SPECIFICATION_NAME, $key);
```

Comment:
Putting assertions here is making real hard to understand the flow of the test.
Try to implement here a simple in-memory cache that stores cached items in an array.
That way you can always set items and retrieve the cached items in the test and verify their properties.
Something like:
```php
public function setUp()
{
    $this->cache = new class () implements TagAwareCacheInterface {...}
}

public function testSomethingSavedInCache()
{
    $specificationLoader = new SpecificationLoader(..., $this->cache, ...);
    $this->specificationLoader->load(...);
    $cachedItems = $this->cache->getCachedItems();
    Assert::assertSomethingOnCacheItems($cacheItems);
}

public function testSomethingFetchedFromCache()
{
    $this->cache->set(...);
    $specificationLoader = new SpecificationLoader(..., $this->cache, ...);
    $this->specificationLoader->load(...);
}
````